openfisca_core/simulations.py

```diff
@@ -44,7 +44,7 @@ def __init__(
         self.tax_benefit_system = tax_benefit_system
         assert tax_benefit_system is not None
 
-        self.entities = entities_instances
+        self.entities = populations
```

Comment:
Mostly because that was going to impact lots of files, making the PR noisier. I agree that it makes sense to at least rename them in Simulation, I've pushed a commit doing that.

The stopping point in any large-scale refactoring is somewhat arbitrary, and this is no exception; there's lots of other places where we were using the term "entity" in an ambiguous way, and it may take a little while before we're using the terms in a completely consistent manner.

In many places we also need to improve encapsulation or move responsibilities around, for instance in `test_runner` we have this:
```Python
if simulation.populations.get(key):
```
where what we really mean is:
```Python
if simulation.entity_exists(key):
```

It's easy to see that the implementation of `entity_exists` is near-trivial, but in the absence of this encapsulation the population/entity ambiguity has "leaked" more than it should have. (Also, as is now obvious, we should be asking the TaxBenefitSystem, not the Simulation.)